Google.Ads.Gax/src/Config/AdsConfig.cs

```diff
@@ -479,6 +479,37 @@ public override void ReadSettings(Dictionary<string, string> settings)
             ReadProxySettings(settings);
         }
 
+        /// <summary>
+        /// Loads the OAuth2 secrets from a stream reader. Note that the stream reader will not be
+        /// disposed of.
+        /// </summary>
+        /// <param name="reader"> A stream reader.</param>
+        public void LoadOAuth2SecretsFromStream(StreamReader reader)
+        {
+            try
+            {
+                string contents = reader.ReadToEnd();
+                Dictionary<string, string> config =
+                    JsonConvert.DeserializeObject<Dictionary<string, string>>(contents);
+
+                ReadSetting(config, oAuth2ServiceAccountEmail);
+                if (string.IsNullOrEmpty(this.OAuth2ServiceAccountEmail))
+                {
+                    throw new ApplicationException(ErrorMessages.ClientEmailIsMissingInJsonFile);
```

Comment:
Nit: Both error messages mention "JSON secrets file", but now you're using those errors for files and streams

Reply:
Good point, addressed!